internal/common/util.go

```diff
@@ -0,0 +1,16 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func ReportConversionError(header string, subheader string, value any, diags *diag.Diagnostics) {
```

Comment:
I don't think I see this being used anywhere outside of this package does this need to be made public?

```suggestion
func reportConversionError(header string, subheader string, value any, diags *diag.Diagnostics) {
```

Reply:
I thought it was based on it being used outside the file, but was wrong about that. I could change it back to lowercase

Reply:
It's about being used outside of the package. 👍